### ISSUE TITLE
Verify that the API runs on linux

### ISSUE THREAD

**Issue by niemyjski** [exceptionless/Exceptionless#66]
This will help us in a few scenarios

1. Migration to vnext
2. Docker images ( #64 )
3. Cheaper hosting for everyone :+1:

**Comment by niemyjski** [comment]
I got the following error running with the latest version of mono:

```
Application Exception
System.MissingMethodException
Method not found: 'System.Web.HttpApplication.RegisterModule'.

Description: HTTP 500.Error processing request.
Details: Non-web exception. Exception origin (name of application or object): mscorlib.
Exception stack trace:
  at (wrapper managed-to-native) System.Reflection.MonoMethod:InternalInvoke (System.Reflection.MonoMethod,object,object[],System.Exception&)
  at System.Reflection.MonoMethod.Invoke (System.Object obj, BindingFlags invokeAttr, System.Reflection.Binder binder, System.Object[] parameters, System.Globalization.CultureInfo culture) [0x00054] in /private/tmp/source-mono-mac-3.12.99-branch-32/bockbuild-mono-3.12.0-branch/profiles/mono-mac-xamarin/build-root/mono-3.12.1/mcs/class/corlib/System.Reflection/MonoMethod.cs:230 
Version Information: 3.12.1 ((detached/b7764aa Fri Mar 6 15:32:47 EST 2015); ASP.NET Version: 4.0.30319.17020
```

**Comment by niemyjski** [comment]
https://bugzilla.xamarin.com/show_bug.cgi?id=21810

**Comment by Nonobis** [comment]
@niemyjski : this bug has been fixed in the lastest mono version ... Maybe there is hope ?

Andres G. Aragoneses 2015-06-09 10:03:09 EDT

This has been fixed in mono master (
https://github.com/mono/mono/commit/58e0608c10c621b15fd9f396f8e46c567891b231 ),
please mark the bug as RESOLVED->FIXED.

**Comment by niemyjski** [comment]
@Nonobis yeah, I'll try once this has been pushed in a mono update (or has it already been pushed).

**Comment by niemyjski** [comment]
@Nonobis can you try and see if it's fixed?

**Comment by Nonobis** [comment]
@niemyjski : yes can try it, do you have a docker package to test ? 
I don't have a linux actually, only a synology nas i don't know if this will work on this environnement.

**Comment by niemyjski** [comment]
We currently don't have a docker package to test. I'm going to also pull down latest on my mac and try it

**Comment by niemyjski** [comment]
Looks like there still is issues with mono 4.0.2.

```
System.MissingMethodException
Method 'HttpApplication.RegisterModule' not found.

Description: HTTP 500.Error processing request.
Details: Non-web exception. Exception origin (name of application or object): mscorlib.
Exception stack trace:
  at (wrapper managed-to-native) System.Reflection.MonoMethod:InternalInvoke (System.Reflection.MonoMethod,object,object[],System.Exception&)
  at System.Reflection.MonoMethod.Invoke (System.Object obj, BindingFlags invokeAttr, System.Reflection.Binder binder, System.Object[] parameters, System.Globalization.CultureInfo culture) [0x00054] in /private/tmp/source-mono-mac-4.0.0-branch-c5sr2/bockbuild-mono-4.0.0-branch/profiles/mono-mac-xamarin/build-root/mono-4.0.2/mcs/class/corlib/System.Reflection/MonoMethod.cs:230 
Version Information: 4.0.2 ((detached/c99aa0c Thu Jun 11 18:53:01 EDT 2015); ASP.NET Version: 4.0.30319.170
```

**Comment by niemyjski** [comment]
https://bugzilla.xamarin.com/show_bug.cgi?id=21810

**Comment by niemyjski** [comment]
They say it's fixed in master but the latest release is 4.0.2 and that's newer than when it was committed to master.. So I'm not sure.. maybe it will be in the next release..

**Comment by niemyjski** [comment]
I'm closing this as we plan to port to coreclr in the near future (when our dependencies have been updated).
